app/src/main/java/com/example/mystore/ui/main/MainActivity.kt

```diff
@@ -51,12 +52,25 @@ class MainActivity : ComponentActivity() {
                                     modifier = Modifier.padding(it),
                         ) {
                             composable(route = "Login") {
-                                LoginScreen(navigateToProductDetailsScreen = {
-                                    navController.navigate("ProductDetailsScreen")
+                                LoginScreen(navigateToHomeScreen = {
+                                    navController.navigate("HomeScreen")
                                 })
                             }
-                            composable(route = "ProductDetailsScreen") {
-                                ProductDetailsScreen()
+                            composable(route = "HomeScreen") {
+                                HomeScreen(navigateToProductDetailsScreen = { productId ->
+                                    navController.navigate(
+                                        "ProductDetailsScreen/{productId}".replace(
```

Comment:
The navigation works with `navArgument`s, as shown [here](https://developer.android.com/jetpack/compose/navigation#nav-with-args).